### PR TITLE
Move `developer-hub` to `polkadot-sdk-docs`

### PR DESCRIPTION
**Pull request by juangirini**
This PR is a continuation of https://github.com/paritytech/polkadot-sdk/pull/2102 and part of an initiative started here https://hackmd.io/@romanp/rJ318ZCEp
What has been done:
- The content under `docs/*` (with the exception of `docs/mermaid`) has been moved to `docs/contributor/`
- Developer Hub has been renamed to Polkadot SDK Docs, and the crate has been renamed from `developer-hub` to `polkadot-sdk-docs`
- The content under `developer-hub/*` has been moved to `docs/sdk`

---
Original PR https://github.com/paritytech/polkadot-sdk/pull/2565, it has been close due to too many rebase conflicts

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4590218

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4590490

**Review by franciscoaguirre — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]
Only comment from be would be if the folder name `contributor` is ideal. So far we just used it as dump for all docs.  
Maybe `repo` or `maintain` or something?

File: docs/sdk/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+[package]
+name = "polkadot-sdk-docs"
+description = "The one stop shop for developers of the polakdot-sdk"
+license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
+homepage = "paritytech.github.io"
+repository.workspace = true
+authors.workspace = true
+edition.workspace = true
+# This crate is not publish-able to crates.io for now because of docify.
```
[the remark this answers is not in the document]
Comment (franciscoaguirre) [reply docs/sdk/Cargo.toml:9]: As long as docify still uses git urls, then that's the case

**Review by AndWeHaveAPlan — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Review by liamaharon — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -29,6 +29,7 @@ bin/node-template/Cargo.lock
 nohup.out
 polkadot_argument_parsing
 polkadot.*
+!docs/sdk/src/polkadot_sdk/polkadot.rs
```
Comment (liamaharon) [inline .gitignore:32]: what's this?

Comment (juangirini) [reply .gitignore:32]: line 31 was excluding that file from git, so I added that line to not exclude it

**Pull request merged**